integration-tests/test-cases/scenario-tests/src/test/java/io/cellery/integration/scenario/tests/ObservabilityDashboard.java

```diff
@@ -0,0 +1,365 @@
+        clickOnObservabilityButton(sequenceDiagramButtonXPath);
+        checkSeqDiagram();
+
+        String dependancyDiagramButton = "//*[@id=\"root\"]/div/main/div[3]/div[1]/div/div/div/button[3]";
+        clickOnObservabilityButton(dependancyDiagramButton);
+
+        String dependancyDiagramXPath = "//*[@id=\"root\"]/div/main/div[3]/div[2]/div/div/div/canvas";
+        webDriver.findElement(By.xpath(dependancyDiagramXPath));
+
+    }
+
+    private void checkSeqDiagram() {
+        for (Map.Entry<String, Integer> entry : diagram.getSequenceDiagramCalls().entrySet()) {
+            checkSeqDiagramCell(entry.getKey(), entry.getValue());
+        }
+        for (Map.Entry<String, Integer> entry : diagram.getSequenceDiagramCells().entrySet()) {
+            webDriver.findElement(By.xpath("//*[name()='svg']/*[name()='g'][" + entry.getValue() +
+                    "]/*[name()" +
+                    "='text']")).click();
+            String firstComp =
+                    webDriver.findElement(By.xpath("//*[name()='svg']/*[name()='g'][2]/*[name()" +
+                            "='text']/*[name()" +
+                            "='tspan']")).getText();
+            Assert.assertEquals(firstComp, entry.getKey());
+        }
+
+    }
+
+    private void checkSeqDiagramCell(String instanceName, int cellSlotNumber) {
+        String cellName =
+                webDriver.findElement(By.xpath("//*[name()='svg']/*[name()='g'][" + cellSlotNumber +
+                        "]/*[name()" +
+                        "='text']/*[name()" +
+                        "='tspan']")).getText();
+        Assert.assertEquals(cellName, instanceName.replace("-", "_"));
+    }
+
+    public void loginObservability() {
+        webDriver.get(Constants.DEFAULT_OBSERVABILITY_URL);
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("/html/body/header/div/div/a/img")));
+        WebElement username = webDriver.findElement(By.id("username"));
+        WebElement password = webDriver.findElement(By.id("password"));
+        username.sendKeys("admin");
+        password.sendKeys("admin");
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//*[@id=\"loginForm\"]/div[6" +
+                "]/div/button"))).click();
+        try {
+            wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("approveCb"))).click();
+            wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("consent_select_all"))).click();
+            wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("approve"))).click();
+        } catch (Exception ignored) {
+
+        }
+        String personalInfoHeader = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath(
+                "//*[@id=\"root" +
+                "\"]/div/header/div/h6"))).getText();
+        Assert.assertEquals(personalInfoHeader, Constants.OBSERVABILITY_WEB_CONTENT, "Cellery Observability" +
```

Comment:
Shall we move the assertion to the test method since it is a part of the test?

Reply:
This applies to all other methods in this class

Reply:
If we are doing this, we will have to repeat code for the number of component each cell has.However since all the UI components are same in observability portal we can avoid the code repetition this way.